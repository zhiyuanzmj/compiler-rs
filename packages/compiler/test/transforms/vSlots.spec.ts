import { describe, expect, test } from 'vitest'
import {
  transformChildren,
  transformElement,
  transformText,
  transformVBind,
  transformVOn,
  transformVSlot,
  transformVSlots,
} from '../../src'
import { makeCompile } from './_utils'

const compileWithSlots = makeCompile({
  nodeTransforms: [
    transformElement,
    transformText,
    transformVSlot,
    transformChildren,
  ],
  directiveTransforms: {
    bind: transformVBind,
    on: transformVOn,
    slots: transformVSlots,
  },
})

describe('compiler: transform v-slots', () => {
  test('basic', () => {
    const { code } = compileWithSlots(
      `<Comp v-slots={{ default: ({ foo })=> <>{ foo + bar }</> }}> </Comp>`,
    )
    expect(code).toMatchSnapshot()
  })
})
