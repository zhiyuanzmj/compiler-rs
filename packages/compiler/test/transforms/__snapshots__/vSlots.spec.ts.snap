// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compiler: transform v-slots > basic 1`] = `
"
  const n1 = _createComponent(Comp, null, {
    $: [
      { default: ({ foo })=> <>{ foo + bar }</> }
    ]
  }, true)
  return n1
"
`;
